### DISCUSSION TITLE
In "Calculating Liquidity", why pick the smaller L among the two calculated values?

### DISCUSSION THREAD

**Discussion by uniyj** [Jeiwan/uniswapv3-book#23] [Milestone 1. First Swap] [answered]
Would be useful to have a brief explanation of why the smaller L value is used. Thanks.

**Comment by 0xtito** [comment]
Ya I am having troubling understanding why picking the smaller L would distribute the liquidity evenly along the curve while the bigger L wouldn't.

**Comment by uniyj** [comment, marked as answer]
This is now addressed in https://github.com/Jeiwan/uniswapv3-book/commit/ed077b0b9b8e905d5c3cbef232f7d5aa9f4bf52d
: )

**Comment by 1NT18IS080** [comment]
What do we mean, when we say
_Because the bigger one already includes the smaller one!_ 
A bit more explanation will be very helpful

**Reply by uniyj** [reply]
If you follow the rest of the "Calculating Liquidity" chapter, you will see that you are calculating two `L`s for the delta-x and delta-y that you are looking to add. Due to rounding errors, the two `L`s will not be exactly equal. The liquidity you are adding (within the price range defined) is distributed evenly, and so the two `L`s actually **need to be equal**. In the example given, with the intended delta-x and delta-y, the `L` calculated from delta-x is the bigger value. If you choose to use that bigger `L`, you will then need to add more delta-y than intended, to achieve that bigger `L`. But if you choose to use the smaller `L` (calculated from delta-y), then the actual delta-x needed is slightly less, and the leftover can just be unused. It's easier to have some leftover of delta-x by adding the smaller `L`, instead of having to calculate and add extra delta-y to achieve the bigger `L`.

Think of it like you have $10, and it's easier to buy something at $9.90 and get $0.10 change back, instead of trying to buy something at $10.10 and thus having to go add an extra $0.10 before you can do the purchase.

**Reply by TiagoSantanaSilva** [reply]
Thank you for the answer! Very elucidating
What do you mean by "left-over" does that delta remain in the liquidity provider's wallet just like $0.10 change is given back to me from my purchase?
